packages/flutter_tools/gradle/flutter.gradle

```diff
@@ -395,13 +477,24 @@ class FlutterPlugin implements Plugin<Project> {
             extraGenSnapshotOptionsValue = project.property('extra-gen-snapshot-options')
         }
 
-        Boolean buildSharedLibraryValue = this.getBuildShareLibrary(project)
-        String targetPlatformValue = this.getTargetPlatform(project)
-
+        def targetPlatforms = this.getTargetPlatforms(project)
         def addFlutterDeps = { variant ->
-            String flutterBuildMode = buildModeFor(variant.buildType)
+            if (this.splitPerAbi(project)) {
+                variant.outputs.each { output ->
+                    // Assigns the new version code to versionCodeOverride, which changes the version code
+                    // for only the output APK, not for the variant itself. Skipping this step simply
+                    // causes Gradle to use the value of variant.versionCode for the APK.
+                    // For more, see https://developer.android.com/studio/build/configure-apk-splits
+                    def abiVersionCode = ABI_VERSION.get(output.getFilter(OutputFile.ABI))
+                    if (abiVersionCode != null) {
+                        output.versionCodeOverride =
+                            abiVersionCode * 1000 + variant.versionCode
```

Comment:
This is going to prevent people from transparently controlling the version code of their APKs. I don't think it's a breaking change exactly but it may impact developers who are already relying on a specific versioning scheme like https://developer.android.com/google/play/publishing/multiple-apks.html#using-a-version-code-scheme, or who are relying on their builds matching a particular version code that's been uploaded to the Play Developer Console. Can there be some documentation about this where the flag is defined so people aren't surprised?

Reply:
Yep. It's documented in `flutter help build apk`. It's an optional flag.